codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowControllerTest.java

```diff
@@ -1050,6 +1050,10 @@ public void error(Throwable t) {
         }
 
         @Override
+        public void writeComplete() {
```

Comment:
Ad 2) That's actually a very good test case. Turns out we currently call error() couple of times if error() throws an Exception.
Ad 3) AFAIK if `writeComplete` throws an Exception we must not call `error` as we state in the docs that the flow controller will either call `error` or `writeComplete` but not both.